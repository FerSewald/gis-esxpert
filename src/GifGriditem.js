import React from 'react'
import PropTypes from 'prop-types'; 


export const GifGriditem = ({title, url}) => {

    return (
        <div className="card animate__animated animate__fadeIn">
            <img src={url} alt={title} ></img>
            <p>{title}</p>
        </div>
    )
}
GifGriditem.ProtoTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}