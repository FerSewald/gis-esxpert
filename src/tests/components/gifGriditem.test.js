import React from 'react';
import { GifGriditem } from '../../GifGriditem';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';



describe('Pruevas en <GifGriditem />', () => {
    const title = 'un titulo';
    const url = 'https://logalhost/algo.gif'
    const wrapper = shallow(<GifGriditem title={title} url={url}/>);

    test('debe retornar componentes correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de tener un parrafo con el title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);

    });

    test('debe tener una img igual a url', () => {
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);

    });

    test('debe de tener animate__fadeIn', () => {
        const div = wrapper.find('div');
        const clasName = div.prop('className');
        expect(clasName.includes('animate__fadeIn')).toBe(true);
    })
    
    

});
