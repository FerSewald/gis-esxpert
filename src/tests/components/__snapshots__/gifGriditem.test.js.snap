// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruevas en <GifGriditem /> debe retornar componentes correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="un titulo"
    src="https://logalhost/algo.gif"
  />
  <p>
    un titulo
  </p>
</div>
`;
